### PR TITLE
Filtered fixes from Hurdlr

### PR DESCRIPTION
**Pull request by aaustin**
Filtered out the fixes from https://github.com/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/pull/118

### CODE REVIEW

**Comment by aaustin** [comment]
@renesansz @jestoniyap Please review and confirm alright.

File: src/android/io/branch/BranchSDK.java [hunk 1]
```diff
@@ -226,19 +219,11 @@ public boolean execute(String action, JSONArray args, CallbackContext callbackCo
...
         this.instance = Branch.getAutoInstance(this.activity.getApplicationContext());
 
-        if (callbackContext == null) {
-            this.instance.initSession(new SessionListener(), activity.getIntent().getData(), activity);            
-        } else {
-            this.instance.initSession(new SessionListener(callbackContext), activity.getIntent().getData(), activity);
-        }
-
-
+        this.instance.initSession(new SessionListener(callbackContext), activity.getIntent().getData(), activity);
```
Comment (renesansz) [inline src/android/io/branch/BranchSDK.java]: The if/else statement should not be removed.
This was a fixed for `initSession` for `onNewIntent`condition, wherein it doesn't have a callback instance being passed.

See this [PR](https://github.com/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/pull/103/files)

Comment (aaustin) [inline src/android/io/branch/BranchSDK.java]: I fixed the initSession code to properly handle null callbackContext instead. Did it throw errors?

Comment (renesansz) [inline src/android/io/branch/BranchSDK.java]: Yes, it would throw an error that the `callbackContext` parameter has an invalid value `null`.

Comment (aaustin) [inline src/android/io/branch/BranchSDK.java]: Look at the new code on line 701-721. I moved the check for null there.

**Comment by aaustin** [comment]
@renesansz Can you please test and confirm with the testbed that all compiles correctly here?

**Comment by aaustin** [comment]
on Android and iOS

**Comment by renesansz** [comment]
@aaustin , sure, will test out the PR. 👍

**Comment by renesansz** [comment]
@aaustin , I can't test out much for the iOS because I only have an emulator.
Will ask @jestoniyap to test out the iOS.

**Comment by renesansz** [comment]
@aaustin  Got this error upon building for iOS.
```
..defaultCenter] postNotification:[NSNotification notificationWithName:@"BSDKPostUnhandledURL" object:[url absoluteString]];
```

**Comment by aaustin** [comment]
ack! oops - try now?

**Comment by renesansz** [comment]
@aaustin  new errors :(
```
/Users/renesansz/Workspace/Ingenuity/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/src/ios/BranchSDK.m:379:26: error:
      variable is not assignable (missing __block type specifier)
            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:params];
            ~~~~~~~~~~~~ ^
/Users/renesansz/Workspace/Ingenuity/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/src/ios/BranchSDK.m:381:26: error:
      variable is not assignable (missing __block type specifier)
            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:[error localizedDescription]];
            ~~~~~~~~~~~~ ^
/Users/renesansz/Workspace/Ingenuity/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/src/ios/BranchSDK.m:514:50: warning:
      undeclared selector 'absoluteString:' [-Wundeclared-selector]
    if ([notification.object respondsToSelector:@selector(absoluteString:)]) {
                                                 ^
/Users/renesansz/Workspace/Ingenuity/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/src/ios/BranchSDK.m:527:10: error:
      missing '[' at start of message send expression
        [self.commandDelegate evalJs:[NSString stringWithFormat:@"NonBranchLinkHandler(%@)", resultString]];
         ^
         [
/Users/renesansz/Workspace/Ingenuity/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/src/ios/BranchSDK.m:527:108: error:
      expected ']'
        [self.commandDelegate evalJs:[NSString stringWithFormat:@"NonBranchLinkHandler(%@)", resultString]];
                                                                                                           ^
/Users/renesansz/Workspace/Ingenuity/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/src/ios/BranchSDK.m:527:9: note:
      to match this '['
        [self.commandDelegate evalJs:[NSString stringWithFormat:@"NonBranchLinkHandler(%@)", resultString]];
        ^
1 warning and 4 errors generated.
```

**Comment by aaustin** [comment]
more typos - fixed now.. hopefully once and for all!

**Comment by renesansz** [comment]
@aaustin ,
cool! works well so far 👍

**Pull request merged**